src/dynprog.rs

```diff
@@ -108,6 +112,58 @@ where
         script
     }
 
+    /// Return all the edit scripts to rewrite the source sequence to the
+    /// target sequence. If you want just one edit script, use the
+    /// `edit_script` method instead.
+    pub fn edit_scripts(&self) -> HashSet<Vec<IndexedOperation<M::Operation>>> {
+        // Find all scripts that lead to the lowest edit distance using
+        // breadth-first search.
+
+        // Start the search in the lower-right corner with the final cost.
+        let mut q: VecDeque<BacktrackState<M, T>> = VecDeque::new();
+        q.push_back(BacktrackState {
+            source_idx: self.pair.source.len(),
+            target_idx: self.pair.target.len(),
+            script: Vec::new(),
+        });
+
+        let mut scripts = HashSet::new();
+        while let Some(BacktrackState {
+            source_idx,
+            target_idx,
+            script,
+        }) = q.pop_front()
+        {
+            // Process all operations/origins that can lead to the current cell's
+            // cost.
+            for op in self.measure
+                .backtracks(&self.pair, &self.cost_matrix, source_idx, target_idx)
+            {
+                let (new_source_idx, new_target_idx) =
+                    op.backtrack(&self.pair, source_idx, target_idx)
+                        .expect("Cannot backtrack");
+                let mut new_script = script.clone();
+
+                new_script.push(IndexedOperation::new(op, new_source_idx, new_target_idx));
+
+                if new_source_idx == 0 && new_target_idx == 0 {
+                    // If we are in the upper-left cell, we have a complete script.
+                    new_script.reverse();
+                    scripts.insert(new_script);
+                } else {
+                    // Otherwise, add the state to the queue to explore later.
+                    q.push_back(BacktrackState {
+                        source_idx: new_source_idx,
+                        target_idx: new_target_idx,
+                        script: new_script,
+                    })
+                }
+            }
+        }
+
```

Comment:
Should there be a 3rd else-if case in addition to
```if new_source_idx == 0 && new_target_idx == 0 {``` (line 149)
and
```} else {``` (line 153)
along the lines of the `assert_eq!` in the `edit_script` function if no backtracking to cell (0,0) is possible, or would a state in _all_ such cases be added to `q`?

Reply:
I will add an assertion here, but it's different than the earlier one. It should assert that `script` is non-empty.

I think it should be an assertion (and not a regular if-clause), because no alignment can be constructed in which backtracking to (0, 0) is not possible. `Align::align` panics in such cases.